packages/rules.lua

```diff
@@ -31,23 +67,80 @@ local function registerCommands (_)
     })
   end, "Draws a blob of ink of width <width>, height <height> and depth <depth>")
 
+  SILE.registerCommand("hrulefill", function(options, _)
+    local raise
+    local thickness
+    if options.position and options.raise then
+      SU.error("hrulefill cannot have both position and raise parameters")
+    end
+    if options.thickness then
+      thickness = SU.cast("measurement", options.thickness)
+    end
+    if options.position == "underline" then
+      local underlinePosition, underlineThickness = getUnderlineParameters()
+      thickness = thickness or underlineThickness
+      raise = underlinePosition
+    elseif options.position == "strikethrough" then
+      local yStrikeoutPosition, yStrikeoutSize = getStrikethroughParameters()
+      thickness = thickness or yStrikeoutSize
+      raise = yStrikeoutPosition + thickness / 2
+    elseif options.position then
+      SU.error("Unknown hrulefill position '"..options.position.."'")
+    else
+      raise = SU.cast("measurement", options.raise or "0")
+    end
+
+    SILE.typesetter:pushExplicitGlue(hrulefillglue({
+      raise = raise,
+      thickness = thickness or SILE.measurement("0.2pt"),
+    }))
+  end, "Add a huge horizontal hrule glue")
+
   SILE.registerCommand("fullrule", function (options, _)
-    SILE.call("raise", { height = options.raise or "0.5em" }, function ()
-      SILE.call("hrule", {
-          height = options.height or "0.2pt",
-          width = options.width or "100%lw"
+    local thickness = SU.cast("measurement", options.thickness or "0.2pt")
+    local raise = SU.cast("measurement", options.raise or "0.5em")
+
+    -- BEGIN DEPRECATION COMPATIBILITY
+    if options.height then
+      SU.deprecated("\\fullrule[…, height=…]", "\\fullrule[…, thickness=…]", "0.13.0", "0.14.0")
+      thickness = SU.cast("measurement", options.height)
+    end
+    if not SILE.typesetter:vmode() then
+      SU.deprecated("\\fullrule in horizontal mode", "\\hrule or \\hrulefill", "0.13.0", "0.14.0")
+      if options.width then
+        SU.deprecated("\\fullrule with width", "\\hrule and \\raise", "0.13.0", "0.14.0")
+        SILE.call("raise", { height = raise }, function ()
+          SILE.call("hrule", {
+            height = thickness,
+            width = options.width
+          })
+        end)
+      else
+        -- This was very broken anyway, as it was overflowing the line.
+        -- At least we try better...
+        SILE.call("hrulefill", { raise = raise, thickness = thickness })
+      end
+     return
+    end
+    if options.width then
+      SU.deprecated("\\fullrule with width", "\\hrule and \\raise", "0.13.0", "0.14.0")
```

Comment:
```suggestion
      SU.deprecated("\\fullrule with width", "\\hrule and \\raise", "0.13.1", "0.14.0")
```

We can't deprecate retrospectively, the first version argument should be the version where we start warning and the second one where we stop shimming and error.

I would prefer to keep shimming at least one major version cycle when feasible. Not it was *not* feasible to do with the stdlib / document class stuff so that is set to error after just one major version.

Reply:
If there isn't an urgent *need* to have an earlier EOL for the shim, lets make it 2 major bumps by default, so v0.15.0. If there is some reason to hurry to drop the shim (e.g. if some other refatoring is waiting on it being completely deprecated) then 1 major version is fine by me.

Reply:
I fully agree on the "2 major bumps" - so re-targeted to 0.15.0.
Done.